app/src/webview/java/org/mozilla/focus/webview/TrackingProtectionWebViewClient.java

```diff
@@ -56,15 +66,19 @@ protected Void doInBackground(Void... voids) {
         // background loading of the lists as early as possible.
         triggerPreload(context);
 
-        this.blockingEnabled = true;
+        final Settings settings = Settings.getInstance(context);
+        settings.getPreferences().registerOnSharedPreferenceChangeListener(this);
```

Comment:
Does this listener go away if the webview is destroyed? (Could this cause a crash?) Do we need to unregister it?

Reply:
Actually, looks like it's a weak reference. Can't hurt to unregister the listener though.
https://developer.android.com/reference/android/content/SharedPreferences.html#registerOnSharedPreferenceChangeListener(android.content.SharedPreferences.OnSharedPreferenceChangeListener)